text/0017-message/0017-message.md

```diff
@@ -0,0 +1,181 @@
+
+
+### Message ID
+The message ID will be the `BLAKE2b-256` hash of the byte contents of the message. It should be used by the nodes to index the messages and by external APIs.
+
+
+### Message Structure
+
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>Version</td>
+        <td>varint</td>
+        <td>The message version. The schema specified in this RFC is for version <strong>1</strong> only. </td>
+    </tr>
+    <tr>
+        <td>Parent1 (<code>trunk</code>)</td>
+        <td>ByteArray[32]</td>
+        <td>The Message ID of the first <i>Message</i> we reference.</td>
+    </tr>
+    <tr>
+        <td>Parent2 (<code>branch</code>)</td>
+        <td>ByteArray[32]</td>
+        <td>The Message ID of the second <i>Message</i> we reference.</td>
+    </tr>
+    <tr>
+        <td>Payload Length</td>
+        <td>varint</td>
+        <td> The length of the Payload. Since its type may be unknown to the node it must be declared in advanced.</td>
+    </tr>
+    <tr>
+        <td colspan="1">
+            Payload
+        </td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Generic Payload</summary>
+                <blockquote>
+                An outline of a general payload
+                </blockquote>
+                <table>
+                    <tr>
+                        <th>Name</th>
+                        <th>Type</th>
+                        <th>Description</th>
+                    </tr>
+                    <tr>
+                        <td>Payload Type</td>
+                        <td>varint</td>
+                        <td>
+                            The type of the payload. It will instruct the node how to parse the fields that follow. Types in the range of 0-127 are "core types" that all nodes are expected to know.
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>Data Fields</td>
+                        <td>ANY</td>
+                        <td>A sequence of fields, where the structure depends on <code>payload type</code>.</td>
+                    </tr>
+                    <tr>
+                        <td>Nested Payload (Optional)</td>
+                        <td> Payload</td>
+                        <td> Some payloads support nesting other payloads inside of them.</td>
+                    </tr>
+                </table>
+            </details>
+    <tr>
+        <td>Nonce</td>
+        <td>uint64</td>
+        <td>The nonce which lets this message fulfill the Proof-of-Work requirement.</td>
+    </tr>
+</table>
+
+### Message PoW Hash
+
+This hash is computed by converting all the messages raw bytes into trytes with `b1t6`. Then hash the result with Curl-P-81. The result will be used to verify the proof-of-work done on the message. This will be used for spam protection.
```

Comment:
We usually put the `t` before the `b`, so `t6b1`.

Reply:
Also, trits, not trytes.

Reply:
I think this is fundamentally different:
- `t5b1` encodes ternary data into binary, by converting each group of 5 trits into one byte; it cannot decode arbitrary binary data
- `b1t6` encodes binary data into ternary, by converting each byte into 6 trits; it cannot decode arbitrary ternary data

In [RFC-15](https://iotaledger.github.io/protocol-rfcs/0015-binary-to-ternary-encoding/0015-binary-to-ternary-encoding.html) it is also called `b1t6`.

Reply:
I think maybe I can delete this entire section. Now we have an RFC dedicated for computing this PoW hash

Reply:
Yes, the entire section should be deleted. See https://github.com/iotaledger/protocol-rfcs/pull/17#discussion_r476267452

Reply:
due to me looking at the RFC while you are commenting, caused me to repeat your remarks